portal-kernel/src/com/liferay/portal/kernel/security/permission/RoleContributor.java

```diff
@@ -26,11 +28,7 @@
  */
 public interface RoleContributor {
 
-	/**
-	 * Contribute to the collection of user roles.
-	 *
-	 * @param roleCollection the managed collection of roles
-	 */
-	public void contribute(RoleCollection roleCollection);
+	public void contribute(
+		User user, UserBag userBag, RoleCollection roleCollection);
```

Comment:
Hi @Preston-Crary,

can we move the `user` and `userBag` into RoleCollection? We discussed a similar thing previously and decided to keep the state in just one place to have clean API.